### DISCUSSION TITLE
Portal 開放學生介接單一簽入(SSO)

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS3073-C/Python-Community-GS3073-C#603] [General]
相信在學校的各位使用不同系或行政單位的網站都經常是透過Portal登入的(如下圖)
<img width="1496" alt="Screenshot 2023-01-06 at 01 46 03" src="https://user-images.githubusercontent.com/32983347/210846192-1ab3f226-4843-490d-8aab-bfb82d60bb7c.png">
這個單一簽入的服務過往只開放給教職員申請使用，但今年在校生也能申請這個服務了，同學可以自已開發網站，並讓使用者透過Portal登入，同時從Portal得到登入者資訊(像是:系所、姓名...)，Portal支援的協定包括OAuth, OpenID, SAML等，可說是市面上常用的協議都有支援，對於同學開發系統也會更方便(據我所知，交大只有開放OAuth協議給學生使用)。

若是對於開發目標客群是校內的應用程式有興趣的同學可以試試看～

申請成為Portal應用程式開發人員(免審核，馬上申請馬上通過): https://portal.ncu.edu.tw/my/apps-dev-member
如何與Portal介接: https://portal.ncu.edu.tw/about/howto
介接範例: https://github.com/ncucc/portal4g-doc

**Comment by awehao** [comment]
竟然還有這麼方便的功能，對我們來說是一大福利诶，但我覺得我暫時應該用不到這個功能

**Comment by number32984** [comment]
太神奇啦，這樣大家就可以開發網站了，非常的方便呢

**Comment by manipulateed** [comment]
想不到還有提供這個功能，但不知道畢業後還能不能繼續使用呢?穴校也真是提供非常多實用的工具，讓學生的學習更加豐富多元呢。
